src/duckling/project/project.service.ts

```diff
@@ -128,6 +140,47 @@ export class ProjectService {
         }
     }
 
+    private async _saveUserMetaData(userMetaData : UserMetaData) : Promise<void> {
+        let json = JSON.stringify(userMetaData, null, 4);
+        let saveResult = await this._jsonLoader.saveJsonToPath(this.getUserMetaDataPath(USER_META_DATA_FILE), json);
+        if (!saveResult.isSuccess) {
+            this._snackbar.addSnack("Error: There was a problem saving last opened map!");
+        }
+    }
+
+    private async _loadUserMetaData() : Promise<UserMetaData> {
+        let fileExists = await this._pathService.pathExists(this.getUserMetaDataPath(USER_META_DATA_FILE));
+        if (fileExists) {
+            let json = await this._jsonLoader.getJsonFromPath(this.getUserMetaDataPath(USER_META_DATA_FILE));
+            let userPreferences = JSON.parse(json);
+            await this._fillMissingUserPreferences(userPreferences);
```

Comment:
We can do this outside of the fileExists condition -- then we can remove the _newUserPreferences() function as well